src/libse/AudioToText/AudioToTextPostProcessor.cs

```diff
@@ -226,46 +232,48 @@ public Subtitle AddPeriods(Subtitle inputSubtitle, string language)
 
             var subtitle = new Subtitle(inputSubtitle);
-            for (var index = 0; index < subtitle.Paragraphs.Count - 1; index++)
+
+            // create default paragraph that will be use as a placeholder for when "next" is not present
+            var nullDuration = subtitle.Paragraphs.Last().StartTime.TotalMilliseconds + SetPeriodIfDistanceToNextIsMoreThan + 1;
+            var defaultParagraph = new Paragraph()
+            {
+                StartTime = new TimeCode(nullDuration),
+                Text = string.Empty
+            };
+
+            for (var index = 0; index < subtitle.Paragraphs.Count; index++)
             {
                 var paragraph = subtitle.Paragraphs[index];
-                var next = subtitle.Paragraphs[index + 1];
-                if (next.StartTime.TotalMilliseconds - paragraph.EndTime.TotalMilliseconds > SetPeriodIfDistanceToNextIsMoreThan &&
-                    !paragraph.Text.EndsWith('.') &&
-                    !paragraph.Text.EndsWith('!') &&
-                    !paragraph.Text.EndsWith('?') &&
-                    !paragraph.Text.EndsWith(',') &&
-                    !paragraph.Text.EndsWith(':'))
+                var next = subtitle.GetParagraphOrDefault(index + 1) ?? defaultParagraph;
+
+                // paragraph is already closed
+                if (paragraph.Text.EndsWith('.') ||
+                    paragraph.Text.EndsWith('!') ||
+                    paragraph.Text.EndsWith('?') ||
+                    paragraph.Text.EndsWith(',') ||
+                    paragraph.Text.EndsWith(':'))
+                {
+                    continue;
+                }
+
+                var timeDistanceToNextLine = next.StartTime.TotalMilliseconds - paragraph.EndTime.TotalMilliseconds;
+                if (timeDistanceToNextLine > SetPeriodIfDistanceToNextIsMoreThan)
+                {
+                    paragraph.Text += ".";
+                }
+                else
                 {
-                    if (next.StartTime.TotalMilliseconds - paragraph.EndTime.TotalMilliseconds > SetPeriodIfDistanceToNextIsMoreThanAlways)
+                    if (TwoLetterLanguageCode == "en" &&
+                        (englishSkipLastWords.Contains(GetLastWord(paragraph.Text)) ||
+                         englishSkipFirstWords.Contains(GetFirstWord(next.Text))))
                     {
-                        paragraph.Text += ".";
+                        continue;
                     }
-                    else
-                    {
-                        var lastWord = GetLastWord(paragraph.Text);
-                        var nextFirstWord = GetFirstWord(next.Text);
-                        if (TwoLetterLanguageCode == "en" && (englishSkipLastWords.Contains(lastWord) || englishSkipFirstWords.Contains(nextFirstWord)))
-                        {
-                            continue;
-                        }
 
-                        paragraph.Text += ".";
```

Comment:
Previously the codes inside this `else` statement will never execute as line 240 would always evaluate to true!